File: 1. Offline 1/Main.cpp
#include <bits/stdc++.h>
#include "Employee.cpp"
using namespace std;

string TakeInput(char*);

//constructor of bank being called twice
//working one minute and then crashing the next

//Created in a way that no bad request is given

int main() {
    Bank bank;
    Account activeAccount;
    Employee activeEmployee;
    int whoIsActiveStatus;

    while (1) {
        char* inputArray;
        string INPUT, inputString = TakeInput(inputArray);
        stringstream ss(inputString);
        ss >> INPUT;

        if (INPUT == "Create") {
            string username;
            int accountType;
            int amount;

            //take username input
            ss >> INPUT; username = INPUT;

            //take accountType input
            ss >> INPUT;
            if (INPUT == "Savings") accountType = SAVINGS;
            else if (INPUT == "Student") accountType = STUDENT;
            else if (INPUT == "Loan") accountType = LOAN;
            else accountType = FIXED;

            //take deposite/loan input
            ss >> INPUT;
            amount = stoi(INPUT);

            Account account(username, accountType, amount);
            bank.addAccount(account);
            whoIsActiveStatus = ACCOUNT;
            activeAccount = account;
        } else if (INPUT == "Deposit") {
            ss >> INPUT;
            int amount = stoi(INPUT);
            activeAccount.Deposit(amount);
        } else if (INPUT == "Withdraw") {
            ss >> INPUT;
            int amount = stoi(INPUT);
            activeAccount.Withdraw(amount, bank.getTime());
        } else if (INPUT == "Query") {
            activeAccount.Query();
        } else if (INPUT == "Request") {
            ss >> INPUT;
            int amount = stoi(INPUT);
            activeAccount.RequestLoan(amount);
        } else if (INPUT == "Open") {
            //ekhaneo jhamela ache, jodi employee open kore
            string username;
            ss >> INPUT;
            username = INPUT;

            whoIsActiveStatus = bank.CheckIfAccountExists(username);

            if (whoIsActiveStatus == ACCOUNT) {
                activeAccount = bank.AccountLogin(username);
            } else {
                //implement employee login/active
            }
            
        } else if (INPUT == "Close") {
            //je active ache, take bank er active variable e rakha lagbe
            // bool whoIsActiveStatus = bank.getWhoIsActive();
            if (whoIsActiveStatus == ACCOUNT) {
                activeAccount.MakeInactive();
                cout << "Transaction Closed for " << activeAccount.getUsername() << "\n";
            } else if (whoIsActiveStatus == EMPLOYEE) {
                //
            } 
            
            whoIsActiveStatus = NOONE;
        }

//Create Alice Student 1000
        
    }
    return 0;
}

string TakeInput(char* inputArray) {
    string inputString;
    cin.getline(inputArray, 100);

    //converts the char array into string
    for (int i=0; inputArray[i]!='\0'; i++) {
        inputString += inputArray[i];
    }
    return inputString;
}

File: 1. Offline 1/Main.java
import java.util.Scanner;

public class Main {
    private static final int ACCOUNT = 1;
    private static final int EMPLOYEE = 2;
    private static final int NOONE = -1;

    public static void main(String[] args) {
        Bank bank = new Bank();
        System.out.println("Bank Created; MD, S1, S2, C1, C2, C3, C4, C5 created");

        Account account = null;
        Employee employee = null;
        int whoIsActiveStatus = NOONE;

        Scanner scanner = new Scanner(System.in);

        while (true) {
            String line = scanner.nextLine();
            String[] command = line.split(" ");

            if (whoIsActiveStatus == NOONE) {
                if (command[0].equalsIgnoreCase("Create")) {
                    if (command[2].equalsIgnoreCase("Savings")) {
                        account = new SavingsAccount(command[1], command[2], Double.parseDouble(command[3]));
                    } else if (command[2].equalsIgnoreCase("Student")) {
                        account = new StudentAccount(command[1], command[2], Double.parseDouble(command[3]));
                    } else if (command[2].equalsIgnoreCase("Fixed")) {
                        account = new FixedDepositAccount(command[1], command[2], Double.parseDouble(command[3]));
                    } else if (command[2].equalsIgnoreCase("Loan")) {
                        account = new LoanAccount(command[1], command[2], Double.parseDouble(command[3]));
                    }

                    bank.AddAccount(account);
                    whoIsActiveStatus = ACCOUNT;
                } else if (command[0].equalsIgnoreCase("Open")) {
                    //
                } else if (command[0].equalsIgnoreCase("INC")) {
                    bank.IncreaseYear();
                }
            } else if (whoIsActiveStatus == ACCOUNT) {
                if (command[0].equalsIgnoreCase("Query")) {
                    account.Query();
                } else if (command[0].equalsIgnoreCase("Withdraw")) {
                    account.Withdraw(Double.parseDouble(command[1]), bank.getYear());
                } else if (command[0].equalsIgnoreCase("Deposit")) {
                    account.Deposit(Double.parseDouble(command[1]));
                } else if (command[0].equalsIgnoreCase("Request")) {
                    account.RequestLoan(Double.parseDouble(command[1]));
                } else if (command[0].equalsIgnoreCase("Close")) {
                    whoIsActiveStatus = NOONE;
                    account.MakeInactive();
                }
            } else if (whoIsActiveStatus == EMPLOYEE) {
                if (command[0].equalsIgnoreCase("Lookup")) {
                    employee.Lookup(command[1]);
                } else if (command[0].equalsIgnoreCase("Approve")) {
                    employee.ApproveLoan();
                } else if (command[0].equalsIgnoreCase("Change")) {
                    employee.ChangeInterestRate(command[1], Double.parseDouble(command[2]));
                } else if (command[0].equalsIgnoreCase("See")) {
                    employee.SeeInternalFund();
                } else if (command[0].equalsIgnoreCase("Close")) {
                    whoIsActiveStatus = NOONE;
                    employee.MakeInactive();
                }
            }
        }
    }
}
